erpnext/controllers/accounts_controller.py

```diff
@@ -1312,7 +1312,7 @@ def validate_quantity(child_item, d):
 			qty_unchanged = prev_qty == new_qty
 			uom_unchanged = prev_uom == new_uom
 			conversion_factor_unchanged = prev_con_fac == new_con_fac
-			date_unchanged = prev_date == new_date if prev_date and new_date else False # in case of delivery note etc
+			date_unchanged = prev_date == getdate(new_date) if prev_date and new_date else False # in case of delivery note etc
```

Comment:
While this does fixe `date_unchanged` behaviour; are you sure this is the reason behind the error you're seeing? 

Because if you change qty then `qty_unchanged` should become False and this check will be irrelevant. 

The error you saw comes from here: https://github.com/frappe/erpnext/blob/fbc163cea66b52494595435200c43ea581fe9bd2/erpnext/controllers/accounts_controller.py#L1827-L1829

Reply:
Code works as expected. Let me explain.

There are 10 lines in the sales order. Other lines have problem with quantities. So throwing the "Cannot set quantity less than delivered quantity" error is OK for them. But, the edited line doesn't have any problem with the date or quantity.

ERPNext was checking the other lines because of date error. This code prevents it.

Reply:
Ok. Merging as this fixes date_unchanged behaviour and prevents unnecessary validations.

Reply:
Need to tinker with some configs to get this ported to develop / v13 branches.